File: GXpert/GXpert.Web/Modules/ServerTypes/Activation/PreDefinedKeyColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EPreDefinedKeyStatus } from "../Rio/Web.Enums.EPreDefinedKeyStatus";
import { PreDefinedKeyRow } from "./PreDefinedKeyRow";

export interface PreDefinedKeyColumns {
    Id: Column<PreDefinedKeyRow>;
    SerialKey: Column<PreDefinedKeyRow>;
    EStatus: Column<PreDefinedKeyRow>;
    InsertDate: Column<PreDefinedKeyRow>;
    InsertUserId: Column<PreDefinedKeyRow>;
    UpdateDate: Column<PreDefinedKeyRow>;
    UpdateUserId: Column<PreDefinedKeyRow>;
    IsActive: Column<PreDefinedKeyRow>;
}

export class PreDefinedKeyColumns extends ColumnsBase<PreDefinedKeyRow> {
    static readonly columnsKey = 'Activation.PreDefinedKey';
    static readonly Fields = fieldsProxy<PreDefinedKeyColumns>();
}

[EPreDefinedKeyStatus]; // referenced types

File: GXpert/GXpert.Web/Modules/Activation/PreDefinedKey/PreDefinedKeyColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Activation.Columns;

[ColumnsScript("Activation.PreDefinedKey")]
[BasedOnRow(typeof(PreDefinedKeyRow), CheckNames = true)]
public class PreDefinedKeyColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string SerialKey { get; set; }
    public short EStatus { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
    public short IsActive { get; set; }
}